### PR TITLE
#695 manipulate delegation and subscription from the modal

### PR DESCRIPTION
**Pull request by abenchi**

### CODE REVIEW

File: frontend/app/calendar-shared-configuration/calendar-shared-configuration.component.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+'use strict';
+
+angular.module('esn.calendar')
+  .component('calendarSharedConfiguration', {
```
Comment (chamerling) [inline frontend/app/calendar-shared-configuration/calendar-shared-configuration.component.js]: `calCalendarSharedConfiguration` here and everywhere

File: frontend/app/calendar-shared-configuration/calendar-shared-configuration.controller.js [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+          return calendar.source.href;
+        });
+
+        return _.filter(userCalendars, function(userCalendar) {
+          return !_.contains(sources, userCalendar.calendar.href);
+        });
+      });
+    }
+
+    function getSubscribedCalendarsForCurrentUser() {
```
Comment (chamerling) [inline frontend/app/calendar-shared-configuration/calendar-shared-configuration.controller.js]: should be a new service, or somewhere in an existing service

File: frontend/app/calendar-shared-configuration/calendar-shared-configuration.controller.js [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+      return calendarHomeService.getUserCalendarHomeId()
+        .then(calendarService.listCalendars)
+        .then(function(calendars) {
+          return userAndExternalCalendars(calendars).publicCalendars || [];
+        });
+    }
+
+    function getPublicCalendarsForUser(user) {
```
Comment (chamerling) [inline frontend/app/calendar-shared-configuration/calendar-shared-configuration.controller.js]: this one also

File: frontend/app/calendar-shared-configuration/calendar-shared-configuration.controller.js [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+        });
+    }
+
+    function getNoResponseDelegationCalendarsForUser(user) {
+      return calendarService.listDelegationCalendars(user._id, 'noresponse')
+        .then(function(delegationCalendars) {
+          return delegationCalendars.map(function(delegationCalendar) {
+            return {
+              calendar: delegationCalendar,
+              type: 'delegation'
```
Comment (chamerling) [inline frontend/app/calendar-shared-configuration/calendar-shared-configuration.controller.js]: constant

File: frontend/app/calendar-shared-configuration/calendar-shared-configuration.controller.js [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+    function _getDelegationCalendars(userCalendars) {
+      return _.filter(userCalendars, { type: 'delegation' });
+    }
+
+    function addSharedCalendars() {
+      $q.all([
+        subscribeToSelectedCalendars(),
+        acceptInvitationToSelectedCalendars()
+      ])
+      .then(function() {
```
Comment (chamerling) [inline frontend/app/calendar-shared-configuration/calendar-shared-configuration.controller.js]: What is bad here is that if one reject we reject all

File: frontend/app/calendar-shared-configuration/calendar-shared-configuration.controller.js (hunk 2 above)
Comment (abenchi) [inline frontend/app/calendar-shared-configuration/calendar-shared-configuration.controller.js]: already discussed on slack

File: frontend/app/calendar-shared-configuration/calendar-shared-configuration.controller.js (hunk 3 above)
Comment (abenchi) [inline frontend/app/calendar-shared-configuration/calendar-shared-configuration.controller.js]: already discussed on slack

**Pull request merged**